apps/aot/src/app/leaderboard/_components/overall-leaderboard.tsx

```diff
@@ -2,16 +2,24 @@ import { Prisma, prisma } from '@repo/db';
 import { ADVENT_CHALLENGE_IDS, LEADERBOARD_RANKING_LIMIT } from '../constants';
 import { DataTableLeaderboard } from '@repo/ui/components/data-table-leaderboard';
 import { overallLeaderboardColumns, type OverallLeaderboardEntry } from './columns';
-import { getCurrentAdventDay } from '~/utils/time-utils';
+import { getCurrentAdventDay, getNextAdventDay } from '~/utils/time-utils';
+import { redisClient } from '@repo/redis';
+
+export const dynamic = 'force-dynamic';
 
 async function getOverallLeaderboard(currentAdventDay: number) {
-  const challengeIdsSoFar = ADVENT_CHALLENGE_IDS.slice(0, currentAdventDay);
+  const cachedRanking = await redisClient.get('aot-overall-leaderboard');
```

Comment:
is this just going to show at the end of the event.

how would this work if we need to update the overall leaderboard day by day. the cache key is static or are we invalidating it somewhere? (Im not done reading the PR)

Reply:
oh nm i see it invalidates at midnight via PXAT

Reply:
> is this just going to show at the end of the event.

Not sure I understand, but to clarify, this leaderboard will always be up-to-date.

Essentially:
1. If we have already gotten 100 unique successful submissions today, it means we have all the data we need to cache the leaderboard.
2. Cache it with expiration at next release date (see code a bit below where I do `redisClient.set`)
3. Once release days hit, we won't be able to hit cache anymore and we will query DB until first 100 unique successful submissions, and then cache again.

Reply:
> how would this work if we need to update the overall leaderboard day by day. the cache key is static or are we invalidating it somewhere? (Im not done reading the PR)

`SET` has a `PXAT` option that will automatically expire the cache key at an exact timestamp.

See docs here: https://redis.io/docs/latest/commands/set/

See diff here: